pybamm/util.py

```diff
@@ -345,3 +345,28 @@ def install_jax(arguments=None):  # pragma: no cover
             f"jaxlib>={JAXLIB_VERSION}",
         ]
     )
+
+# https://docs.pybamm.org/en/latest/source/user_guide/contributing.html#managing-optional-dependencies-and-their-imports
+def have_optional_dependency(module_name, attribute=None):
+    try:
+        # Attempt to import the specified module
+        module = importlib.import_module(module_name)
+
+        if attribute:
+            # If an attribute is specified, check if it's available
+            if hasattr(module, attribute):
+                imported_attribute = getattr(module, attribute)
+                return imported_attribute  # Return the imported attribute
+            else:
+                # Raise an ImportError if the attribute is not available
+                raise ImportError(f"Optional dependency {module_name} is not available. See https://docs.pybamm.org/en/latest/source/user_guide/installation/index.html#optional-dependencies for more details.")
+        else:
+            # Return the entire module if no attribute is specified
+            return module
+
+    except ImportError:
```

Comment:
```suggestion
    except ModuleNotFoundError:
```

`importlib.import_module` raises `ModuleNotFoundError` instead of `ImportError` (I tested this locally)